src/main/java/io/eurora/euroratest/archive/ArchiveService.java

```diff
@@ -0,0 +1,76 @@
+package io.eurora.euroratest.archive;
+
+import io.eurora.euroratest.EuroraTestConfiguration;
+import io.eurora.euroratest.EuroraTestException;
+import io.eurora.euroratest.PathService;
+import io.eurora.euroratest.report.ReportService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import java.util.zip.ZipOutputStream;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ArchiveService {
+
+  private final ZipService zipService;
+  private final UnzipService unzipService;
+  private final FilePool filePool;
+  private final ExecutorService taskExecutor;
+  private final ReportService reportService;
+  private final EuroraTestConfiguration config;
+  private final PathService pathService;
+
+  public void zip(int fileCount) {
+    IntStream.range(0, fileCount).forEach(i -> {
+      ZipOutputStream zipOutputStream = null;
+      try {
+        String filename = "file_" + i;
+        zipOutputStream = makeZipOutputStream(filename);
+        zipService.createArchive(zipOutputStream, filename);
+        zipOutputStream.close();      }
+      catch (Exception e) {
+        log.error("Error {} composing file {}", e.getMessage(), i);
+      }
+    });
+  }
+
+  public void unzipAll() throws EuroraTestException {
+    try {
+      Path unzipFolder = pathService.getUnzipFolder();
+      filePool.loadPool(unzipFolder);
+      reportService.updatePoolFiles(filePool.getSize());
+
+      for (int i = 0; i < config.getDefaultThreadCount(); i++) {
+        taskExecutor.execute(new UnzipTask(filePool, unzipService, reportService, config));
+      }
+      taskExecutor.awaitTermination(1, TimeUnit.SECONDS);
```

Comment:
Let's remove awaitTermination?
Because we shouldn't think that the archiving operation will be completed in less than one second. What if it takes longer? Then your report will not be correct. And if it is executed earlier than one second? Then the thread will be blocked more than necessary.
**I suggest you use CountDownLatch instead of awaitTermination place.**
```suggestion
    countDownLatch.await();
```